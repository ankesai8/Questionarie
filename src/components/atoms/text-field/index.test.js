import React from "react";
import renderer from "react-test-renderer";
import ShallowRenderer from "react-test-renderer/shallow";

import { InputField } from "./index";

const getChildrenByClass = (tree, classString) => {
  const result = tree.root.find(
    (instance) => instance.props.className === classString
  ).props.type;

  return result;
};

describe("InputField", () => {
  const onChange = () => {
    return "function called";
  };

  it("InputField should render properly", () => {

    const renderer = new ShallowRenderer();
    const tree = renderer.render(<InputField type="text" name="name" onChange={onChange} />);
    expect(tree).toMatchSnapshot();
  });
});

describe("InputField  type", () => {
  const onChange = () => {
    return "function called";
  };

  it("should show the InputField  type", () => {
    const tree = renderer.create(<InputField type="email" name="name" onChange={onChange} />);
    const text = getChildrenByClass(tree, "input-field");

    expect(text).toBe("email");

    expect(tree.toJSON()).toMatchSnapshot();
  });

});
