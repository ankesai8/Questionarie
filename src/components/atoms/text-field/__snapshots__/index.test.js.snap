// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField  type should show the InputField  type 1`] = `
<input
  className="input-field"
  name="name"
  onChange={[Function]}
  placeholder="Placeholder text..."
  required={false}
  type="email"
/>
`;

exports[`InputField InputField should render properly 1`] = `
<input
  className="input-field"
  name="name"
  onChange={[Function]}
  placeholder="Placeholder text..."
  required={false}
  type="text"
/>
`;
